moveit_simple/src/moveit_simple.cpp

```diff
@@ -328,27 +347,34 @@ void Robot::execute(const std::string traj_name)
   {
     control_msgs::FollowJointTrajectoryGoal goal;
     goal.trajectory.joint_names = joint_group_->getVariableNames();
-    if ( toJointTrajectory(traj_name, goal.trajectory.points) )
+    try
     {
-      ros::Duration traj_time =
-          goal.trajectory.points[goal.trajectory.points.size()-1].time_from_start;
-      ros::Duration timeout(TIMEOUT_SCALE * traj_time.toSec());
-      if (action_.sendGoalAndWait(goal, timeout) == actionlib::SimpleClientGoalState::SUCCEEDED)
+      if ( toJointTrajectory(traj_name, goal.trajectory.points, collision_check) )
       {
-        ROS_INFO_STREAM("Successfully executed trajectory: " << traj_name);
+        ros::Duration traj_time =
+            goal.trajectory.points[goal.trajectory.points.size()-1].time_from_start;
+        ros::Duration timeout(TIMEOUT_SCALE * traj_time.toSec());
+        if (action_.sendGoalAndWait(goal, timeout) == actionlib::SimpleClientGoalState::SUCCEEDED)
+        {
+          ROS_INFO_STREAM("Successfully executed trajectory: " << traj_name);
+        }
+        else
+        {
+          ROS_ERROR_STREAM("Trajectory " << traj_name << " failed to exectue");
+          throw ExecutionFailureException("Execution failed for "+ traj_name);
+        }
       }
       else
       {
-        ROS_ERROR_STREAM("Trajectory " << traj_name << " failed to exectue");
-        throw ExecutionFailureException("Execution failed for "+ traj_name);
+        ROS_ERROR_STREAM("Failed to convert " << traj_name << " to joint trajectory");
+        throw IKFailException("Conversion to joint trajectory failed for " + traj_name);
       }
     }
-    else
+    catch(CollisionDetected &cd)
     {
-      ROS_ERROR_STREAM("Failed to convert " << traj_name << " to joint trajectory");
-      throw IKFailException("Conversion to joint trajectory failed for " + traj_name);
+      ROS_ERROR_STREAM("Collision detected in the " << traj_name);
+      throw cd;
```

Comment:
It's actually only for logging the traj_name in which the collision has occurred.